### PR TITLE
LPS-137340 Replace input with ImageInput

### PR DESCRIPTION
**Pull request by p2kmgcl**

### CODE REVIEW

**Comment by p2kmgcl** [comment]
ci:test:sf

**Comment by p2kmgcl** [comment]
ci:test:relevant

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/ImageInput.es.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+			{name ? (
+				<ClayForm.Group>
+					<label className="sr-only" htmlFor={imageTitleId}>
+						{Liferay.Language.get('image')}
+					</label>
+
+					<ClayInput.Group small>
+						<ClayInput.GroupItem>
+							<input
+								accept="image/*"
```
Comment (victorg1991) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/ImageInput.es.js:45]: We didn't constraint types in the past, so probably we need to not constraint those now. I wouldn't be surprised if people are using this input for other stuff x)

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/article/small_image.jsp [hunk 2]
```diff
@@ -60,13 +60,33 @@ JournalFileUploadsConfiguration journalFileUploadsConfiguration = (JournalFileUp
...
+			module="js/ImageInput.es"
+			props='<%=
+				HashMapBuilder.<String, Object>put(
+					"name", name
+				).put(
+					"previewURL", previewURL
+				).build()
+			%>'
+		/>
+	</div>
```
Comment (victorg1991) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/article/small_image.jsp:89]: Also, we need to replace it from the templates properties panel: https://github.com/liferay/liferay-portal/blob/48a7bbb3a0f227f78ee9b0ebc26c0816641befbb/modules/apps/journal/journal-web/src/main/resources/META-INF/resources/ddm_template/edit_small_image.jsp#L39-L71. This is the one that the issue mentioned

**Comment by victorg1991** [comment]
Works fine, and code looks solid :). I've left a couple of comments that I think we need to address. Apart from the mentioned above it seems that the result is not the same as in the screenshot attached to the issue: 

Screenshot 2021-09-13 at 08 20 31

Has this been changed?

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/ImageInput.es.js (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/ImageInput.es.js:45]: :disappointed: looked like I good idea, I will change that

**Comment by p2kmgcl** [comment]
> Works fine, and code looks solid :). I've left a couple of comments that I think we need to address. Apart from the mentioned above it seems that the result is not the same as in the screenshot attached to the issue:
> 
> Screenshot 2021-09-13 at 08 20 31
> 
> Has this been changed?

I am not adding the remove button because in the select above there is already a "no image" option, and I didn't want to modify any behavior. But sure I will add some comment to the ticket explaining this. That's also why the selected image is not being shown in the preview until you save.

I think I am writing Mery for the case she wants to update these.

**Comment by p2kmgcl** [comment]
I have talked with Mery and we are keeping the component as it is (just the add/change button, no remove button)

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
Just start reviewing

**Comment by Tim-Cao** [comment]
See https://github.com/liferay-echo/liferay-portal/pull/5883

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**